File: core/src/com/inkostilation/pong/commands/UpdateFieldCommand.java
package com.inkostilation.pong.commands;

import com.inkostilation.pong.engine.IPongEngine;

import java.io.IOException;

public class UpdateFieldCommand<M> extends AbstractPongCommand<M> {
    @Override
    public void execute(IPongEngine engine) throws IOException {
       getEngine().sendFieldState(getMarker());
    }
}

File: core/src/com/inkostilation/pong/commands/UpdateFieldCommand.kt
package com.inkostilation.pong.commands

import com.inkostilation.pong.engine.IPongEngine
import java.io.IOException

class UpdateFieldCommand<M> : AbstractPongCommand<M>() {
    @Throws(IOException::class)
    override fun execute(engine: IPongEngine<*>) {
        getEngine().sendFieldState(marker)
    }
}